scripts/release/release.js

```diff
@@ -0,0 +1,10 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
+const shell = require('shelljs');
+
+const releaseType = process.argv[2] || 'patch';
+
+if (shell.exec(`npm test && npm run compile && npm version ${releaseType} -m "Released version %s" && npm publish && git push`).code !== 0) {
```

Comment:
@bjankord found an issue when running `npm test` inside of shell.exec which allows a false positive.  The pretest script will kick off linting. Linting can fail, but npm test still returns a exit-code of 0 from tests. 

If we take npm test out of shelljs, it will fail on linting errors.